config/clusters/catalystproject-africa/uvri.values.yaml

```diff
@@ -0,0 +1,70 @@
+jupyterhub:
+  ingress:
+    hosts: [uvri.af.catalystproject.2i2c.cloud]
+    tls:
+      - hosts: [uvri.af.catalystproject.2i2c.cloud]
+        secretName: https-auto-tls
+  custom:
+    2i2c:
+      add_staff_user_ids_to_admin_users: true
+      add_staff_user_ids_of_type: "github"
+    homepage:
+      templateVars:
+        org:
+          name: Catalyst Project, Africa - UVRI
+          url: https://2i2c.org
+          logo_url: https://2i2c.org/media/logo.png
+        designed_by:
+          name: "2i2c"
+          url: https://2i2c.org
+        operated_by:
+          name: "2i2c"
+          url: https://2i2c.org
+        funded_by:
+          name: Chan Zuckerberg Initiative - Open Science
+          url: "https://chanzuckerberg.com/science/programs-resources/open-science/"
+  hub:
+    config:
+      JupyterHub:
+        authenticator_class: github
+      GitHubOAuthenticator:
+        oauth_callback_url: https://uvri.af.catalystproject.2i2c.cloud/hub/oauth_callback
+        allowed_organizations:
+          - CatalystProject-Hubs:uvri
+        scope:
+          - read:org
+      # Authenticator:
+      #   admin_users:
+      #     -
+  singleuser:
+    profileList:
+      - display_name: Jupyter
+        slug: jupyter-scipy
+        description: "Python environment"
+        default: true
+        kubespawner_override:
+          image: jupyter/scipy-notebook:2023-06-27
+          default_url: /lab
+        profile_options: &profile_options
+          resource_allocation:
+            display_name: Resource Allocation
+            choices:
+              mem_8:
+                default: true
+                display_name: ~8 GB RAM, ~1.0 CPU
+                kubespawner_override:
+                  mem_guarantee: 6.684G
+                  cpu_guarantee: 0.75
```

Comment:
I'm a bit confused about where these numbers come from. In the issue, it looks like the request was for 4GB / 1CPU.

I suspect these numbers were taken from a hub that had a more extensive list of resource options and are the closest to the resources requested by the community (4GB 1CPU)? Which is a fair assumption, I think, given we don't have docs about the resource allocation script nor guidelines about how to choose these :(

I am not 100% confident to recommend a change here, but would love for @consideRatio's input!  In this particular case, would one use the resource allocation script to generate the numbers or should the exactly requested numbers be used?

Reply:
> In this particular case, would one use the resource allocation script to generate the numbers or should the exactly requested numbers be used?

I'd say no, but I don't know the state of things - its providing requests==limits right? That isn't suitable here I'd say, but of course it can be used to provide something to tweak etc...

Yikes, this is as always a big topic for each single hub.

My proposal is to use this formula and provide options requesting 0.5, 1, 2, and 4 GB of memory, representing ~1/64, ~1/32, ~1/16, and ~1/8 of the node.

- mem request 0.5G
- mem limit 1G (twice the requested memory)
- cpu request 3.6 / 64 (~share of _allocatable_ CPU)
- cpu limit 4 / 64 * 8 (something like eight times the requested CPU excluding the ~400m headroom, but at least 1CPU)

Something like....

```
{
   "mem_guarantee": 0.5G,
   "mem_limit": 1.0G,
   "cpu_guarantee": 0.05625,
   "cpu_limit": 1,
}
{
   "mem_guarantee": 1G,
   "mem_limit": 2G,
   "cpu_guarantee": 0.1125,
   "cpu_limit": 1,
}
{
   "mem_guarantee": 2G,
   "mem_limit": 4G,
   "cpu_guarantee": 0.225,
   "cpu_limit": 2,
}
{
   "mem_guarantee": 4G,
   "mem_limit": 8G,
   "cpu_guarantee": 0.45,
   "cpu_limit": 4,
}
```

Generated from just-in-time writeup:

```python
def scale(n):
    print({
        "mem_guarantee": f"{n}G",
        "mem_limit": f"{n * 2}G",
        "cpu_guarantee": 1/(32/n) * (4-0.4),
        "cpu_limit": max(1, 1/(32/n) * 4 * 16),
    })

scale(0.5)
scale(1)
scale(2)
scale(4)
```